src/cargo/ops/registry.rs

```diff
@@ -134,7 +134,7 @@ fn verify_dependencies(
     registry_src: SourceId,
 ) -> CargoResult<()> {
     for dep in pkg.dependencies().iter() {
-        if dep.source_id().is_path() || dep.source_id().is_git() {
```

Comment:
How come the `is_path` was dropped here? I think we still want this condition, just that some path dependencies have their version automatically filled in?